### PR TITLE
dtkcore@2 2.0.16.1 (new formula)

### PR DESCRIPTION
**Pull request by justforlxz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

I am a dtk maintainer from deepin. This project is based on qt development and encapsulates the basic libraries required for deepin applications.

### CODE REVIEW

**Comment by justforlxz** [comment]
@alebcay hi, I have a question, dtkcore will generate qt prf files and install them in qt's mkspecs/features / directory, but brew forbids me to install to the qt directory. What should I do?

**Comment by SMillerDev** [comment]
Install them to a directory local to the formula. There is no way to write to the directory of other software, nor should there be.

File: Formula/dtkcore@2.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  url "http://packages.deepin.com/deepin/pool/main/d/dtkcore/dtkcore_2.0.16.1.orig.tar.xz"
+  sha256 "79b19a9b01f8a53d2794afa3073d3303b54d6e07da94cc1ef20eb32ea0c3f465"
+
+  depends_on "pkg-config" => :build
+  depends_on "qt"
+
+  patch :DATA
+
+  def install
+    system "qmake", "PREFIX=#{prefix}", "DTK_VERSION=2.0.16.1", ".", "-r"
```
Comment (SMillerDev) [inline Formula/dtkcore@2.rb]: Please use `#{version}` here

File: Formula/dtkcore@2.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
+class DtkcoreAT2 < Formula
+  desc "Deepin Tool Kit Core Module"
+  homepage "https://www.deepin.org"
+  url "http://packages.deepin.com/deepin/pool/main/d/dtkcore/dtkcore_2.0.16.1.orig.tar.xz"
+  sha256 "79b19a9b01f8a53d2794afa3073d3303b54d6e07da94cc1ef20eb32ea0c3f465"
+
+  depends_on "pkg-config" => :build
+  depends_on "qt"
+
+  patch :DATA
```
Comment (SMillerDev) [inline Formula/dtkcore@2.rb]: New formula shouldn't have patches, please request upstream to fix it and make a formula with the new version.

File: Formula/dtkcore@2.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
+class DtkcoreAT2 < Formula
```
Comment (SMillerDev) [inline Formula/dtkcore@2.rb]: We only use `formula@#` if there already is a formula with a different version number. That's not the case here though so you should change this to just be Dtkcore.

**Review by SMillerDev — COMMENTED** [review]
Thanks for the pull request, there's some issues with it that need to be addressed though.

File: Formula/dtkcore@2.rb (hunk 3 above)
Comment (justforlxz) [reply Formula/dtkcore@2.rb]: I see that python@2.rb is written like this, dtkcore@2 is a released version, we will release version 5 in the near future, do I need to submit dtkcore@2.rb at that time and rename the current file to dtkcore.rb?

File: Formula/dtkcore@2.rb (hunk 2 above)
Comment (justforlxz) [reply Formula/dtkcore@2.rb]: dtkcore@2 has been archived and we are developing a new version, so I can only compile by patch.

Comment (SMillerDev) [reply Formula/dtkcore@2.rb]: Then please wait until the new version is released.

File: Formula/dtkcore@2.rb (hunk 3 above)
Comment (SMillerDev) [reply Formula/dtkcore@2.rb]: Yes, the latest version is always without a version number in homebrew. (`python` is version 3 for that reason.)

Comment (justforlxz) [reply Formula/dtkcore@2.rb]: I see, I will wait until the release to submit the update, thank you.

**Pull request closed without merging**